packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3501,25 +3501,24 @@ export class ContainerRuntime
 
 		// eslint-disable-next-line import/no-deprecated
 		const stageControls: StageControlsExperimental = {
-			discardChanges: exitStagingMode(() => {
-				// Pop all staged batches from the PSM and roll them back in LIFO order
-				this.pendingStateManager.popStagedBatches(({ runtimeOp, localOpMetadata }) => {
-					assert(
-						runtimeOp !== undefined,
-						0xb82 /* Staged batches expected to have runtimeOp defined */,
-					);
-					this.rollbackStagedChanges(runtimeOp, localOpMetadata);
-				});
-				this.updateDocumentDirtyState();
-			}),
-			commitChanges: (optionsParam) => {
-				const options = { ...defaultStagingCommitOptions, ...optionsParam };
-				return exitStagingMode(() => {
+			discardChanges: () =>
+				exitStagingMode(() => {
+					// Pop all staged batches from the PSM and roll them back in LIFO order
+					this.pendingStateManager.popStagedBatches(({ runtimeOp, localOpMetadata }) => {
```

Comment:
I removed the below assert after fixing up the typing to ensure we don't get called with undefined.